base/compiler/optimize.jl

```diff
@@ -386,6 +386,31 @@ function inline_worthy(body::Array{Any,1}, src::CodeInfo, sptypes::Vector{Any},
     return true
 end
 
+# Analyzing costs (keep in sync with the above)
+function statement_costs!(cost::Vector{Int}, body::Vector{Any}, src::CodeInfo, sptypes::Vector{Any}, params::OptimizationParams)
+    throw_blocks = find_throw_blocks(body)
+    maxcost = 0
+    for line = 1:length(body)
+        stmt = body[line]
+        thiscost = 0
+        if stmt isa Expr
```

Comment:
The reason we can't just factor this whole thing into a function and use it in both places is to avoid dynamic dispatch?

Reply:
Mostly I thought it would be ugly...either you're storing into an array or summing integers. But I agree that for keeping them synced it would be better to combine them, and I guess a couple of helper functions would make it less ugly.

Reply:
Maybe, it would avoid duplicating the hardcoded 40's though.